src/filters/utils.js

```diff
@@ -1,50 +1,160 @@
-    .replace(/-+$/, ''); // Trim - from end of text
+  if (attrs && attrs.length) {
+    attrs.forEach((attr) => {
+      obj = obj[attr];
+    });
+  }
 
-const unique = (array) =>
-  array.filter((value, index, self) => self.indexOf(value) === index);
+  if (value && obj) {
+    return obj === value || (Array.isArray(obj) && obj.includes(value));
+  }
 
-const jsonString = (item) => JSON.stringify(item);
+  return obj;
+};
+
+/* @docs
+label: just
+category: data
+note: |
+  Returns a filtered array of objects with a given attribute,
+  or the first object where that attribute is equal to a particular value.
+example: |
+  {{ embed.figure(media | just('audio')) }}
+  {{ quotes.blockquote(press | just('slug', 'extension')) }}
+params:
+  array:
+    type: array of objects
+  attrs:
+    type: string | array
+    note: Attribute to filter for
+  value:
+    type: any
+    default: undefined
+    note: Optional attribute-value to just
+*/
+const just = (array, attrs, value) => {
+  const items = array.filter((item) => get(item, attrs, value));
+  return items && value ? items[0] : items;
+};
 
-const only = (list, start, end) => list.slice(start, end);
+/* @docs
+label: getJust
+category: data
+note: |
+  The built-in nunjucks `slice` filter returns
+  an array of arrays at a given length.
+  This filter acts more like the js `array.slice()`,
+  returning a single sub-array from a given start index,
+  to the given end (or end of array).
+example: |
+  {# the first 5 items in a list #}
+  {% for item in my_array | items(0, 5) %}
+params:
+  data:
+    type: object
+  attrs:
+    type: string | array
+    note: The attributes to get
+  test:
+    type: object
+    note: attribute/value pairs to find in the resulting array
+*/
+const getJust = (data, attrs, test) => {
+  data = get(data, attrs);
 
-const get = (data, attr, value) => {
-  if (value) {
-    return data.find((item) => {
-      const hasValue = Array.isArray(item) ? item.includes(value) : false;
-      return item[attr] === value || hasValue;
+  if (data && test) {
+    Object.keys(test).forEach((key) => {
+      data = just(data, key, test[key]);
     });
   }
 
-  return data.filter((item) => item[attr]);
+  return data;
 };
 
+/* @docs
+label: items
+category: data
+note: |
+  The built-in nunjucks `slice` filter returns
+  an array of arrays at a given length.
+  This filter acts more like the js `array.slice()`,
+  returning a single sub-array from a given start index,
+  to the given end (or end of array).
+example: |
+  {# the first 5 items in a list #}
+  {% for item in my_array | items(0, 5) %}
+params:
+  list:
+    type: array
+  start:
+    type: 0-index
+  end:
+    type: 0-index
+*/
+const items = (list, start, end) => list.slice(start, end);
+
+/* @docs
+label: styles
+category: formatting
+note: |
+  Convert an object of property-value pairs
+  into valid CSS for an inline-style.
+example: |
+  <div style="{{ style_dict | styles }}">
```

Comment:
Do we use this anywhere? I imagine it would be helpful for setting certain Custom Properties

Reply:
Yeah, that was the idea. I've mostly been using it in the `embed` macros, as a way to pass in variations.

Reply:
yeah, we use it a few places for exactly that :)